motionplan/tpSpaceRRT.go

```diff
@@ -443,52 +443,50 @@ func (mp *tpSpaceRRTMotionPlanner) getExtensionCandidate(
 	rrt rrtMap,
 	nearest node,
 ) (*candidate, error) {
+	// Get the distance function that will find the nearest RRT map node in TP-space of *this* PTG
+	ptgDistOpt, distMap := mp.make2DTPSpaceDistanceOptions(curPtg)
+
 	nm := &neighborManager{nCPU: mp.planOpts.NumThreads / len(mp.tpFrame.PTGSolvers())}
 	nm.parallelNeighbors = 10
 
 	var successNode node
-	// Get the distance function that will find the nearest RRT map node in TP-space of *this* PTG
-	ptgDistOpt := mp.algOpts.distOptions[curPtg]
 
+	var solution *ik.Solution
 	if nearest == nil {
 		// Get nearest neighbor to rand config in tree using this PTG
-		// TODO: running nearestNeighbor actually involves a ptg.Solve() call, duplicating work.
 		nearest = nm.nearestNeighbor(ctx, ptgDistOpt, randPosNode, rrt)
 		if nearest == nil {
 			return nil, errNoNeighbors
 		}
+
+		rawVal, ok := distMap.Load(nearest.Pose())
+		if !ok {
+			mp.logger.Error("nearest neighbor failed to find nearest pose in distMap")
+			return nil, errNoNeighbors
+		}
+
+		solution, ok = rawVal.(*ik.Solution)
+		if !ok {
+			mp.logger.Error("nearest neighbor ik.Solution type conversion failed")
+			return nil, errNoNeighbors
+		}
+	} else {
+		ptgSolution, err := mp.ptgSolution(curPtg, nearest.Pose(), randPosNode.Pose())
```

Comment:
Doesn't this redo the previous work? "if we have `solution`, then compute a new solution and overwrite `solution`" when we could just use `solution`?

Reply:
I don't see any prior existing solution. the `solution` variable is initialized to a `nil` pointer here: https://github.com/viamrobotics/rdk/pull/3420/files#diff-82a4d2e48ea63f04368d3f5bc9fb16d670e94e76041455dc81d81d9f72b1a35aR454 & is not set until here https://github.com/viamrobotics/rdk/pull/3420/files#diff-82a4d2e48ea63f04368d3f5bc9fb16d670e94e76041455dc81d81d9f72b1a35aR482 if `nearest` is not nil. I'm not seeing the duplicate work you referred to. What am I missing?

Reply:
I think I may have been looking at an old version of the PR because the issue I was seeing definitely no longer exists. Sorry for the false alarm.

Reply:
no problem! It has indeed been through many revisions :)